src/libraries/System.Net.Http/src/System/Net/Http/ByteArrayHelpers.cs

```diff
@@ -22,16 +25,33 @@ internal static bool EqualsOrdinalAsciiIgnoreCase(string left, ReadOnlySpan<byte
                 uint charA = left[i];
                 uint charB = right[i];
 
-                unchecked
+                // We're only interested in ASCII characters here.
+                if ((charA - 'a') <= ('z' - 'a'))
+                    charA -= ('a' - 'A');
+                if ((charB - 'a') <= ('z' - 'a'))
+                    charB -= ('a' - 'A');
+
+                if (charA != charB)
                 {
-                    // We're only interested in ASCII characters here.
-                    if ((charA - 'a') <= ('z' - 'a'))
-                        charA -= ('a' - 'A');
-                    if ((charB - 'a') <= ('z' - 'a'))
-                        charB -= ('a' - 'A');
+                    return false;
                 }
+            }
 
-                if (charA != charB)
+            return true;
+        }
+
+        internal static bool EqualsOrdinalAscii(string left, ReadOnlySpan<byte> right)
```

Comment:
Can always pick up `BytesOrdinalEqualsStringAndAscii` from aspnetcore's `ServerInfrastructure/StringUtilities`

https://github.com/dotnet/aspnetcore/blob/c16d3364d4f83307ac3c6f74de15f8ca96b89759/src/Shared/ServerInfrastructure/StringUtilities.cs#L313-L463

Reply:
Presumably @GrabYourPitchforks will have
https://github.com/dotnet/runtime/pull/34724/files/9d35e5ea0dbf79c474e97c9521f4cb4038beee7b#diff-5643273c2df7053c5ef81c7d9ec49e7cR11-R12
done soon and this can/should just use that.